stdlib/experimental/experimental.flux

```diff
@@ -21,3 +21,20 @@ builtin to
 
 // An experimental version of join.
 builtin join
+
+// Aligns all tables/series to a common start time
+// By default, it aligns to the _start value
+// If _start doesn't exist, it aligns to 1970-01-01T00:00:00Z UTC
+alignTime = (tables=<-, alignTo=time(v: 0)) => {
+  _tableInfo = tables |> tableFind(fn: (key) => true) |> getRecord(idx: 0)
+  _startTime = if alignTo != time(v: 0) then alignTo else if exists _tableInfo._start then _tableInfo._start else time(v: 0)
```

Comment:
Why not instead of computing the _startTime here you do it each time inside the `map` call?

 The way its written it will have to materialize the value of `tables` twice, once to get the startTime and once again to pass through the `stateDuration`, function. Additionally it assumes that the data is grouped by `_start` which may not always be the case.

Reply:
The `_tableInfo` and `_startTime` variables are there to deterministically set the default time to align to if the `alignTo` parameter isn't set. `_tableInfo` just pulls the first record from the first table to provide an idea of the structure of the data. `_startTime` use that info to:

1. Check if the `alignTo` param is set to anything but zero, if so, then
2. Check if the table contains a `_start` column. If so, align to the `_start` value, if not,
3. Align time to the Unix epoch, `time(v: 0)`.

This also makes it so everything aligns to a single `_start`. I don't want to use the `_start` time in each row because windowed data will have a different `_start` value per table.

> Why not instead of computing the _startTime here you do it each time inside the `map` call?

Readability –  but I can do it directly in the `map` call.



The goal here is to